Classes/Issues/GithubClient+Issues.swift

```diff
@@ -183,4 +183,19 @@ extension GithubClient {
         }))
     }
     
+    func setLocked(owner: String, repo: String, number: Int, locked: Bool, completion: @escaping (Result<Bool>) -> ()) {
+        request(Request(
+            path: "repos/\(owner)/\(repo)/issues/\(number)/lock",
+            method: locked ? .put : .delete,
+            completion: { (response, _) in
+                // As per documentation this endpoint returns no content, so all we can validate is that
+                // the status code is "204 No Content".
+                if response.response?.statusCode == 204 {
+                    completion(.success(true))
+                } else {
+                    completion(.error(nil))
```

Comment:
Will it actually never error?

Reply:
I'm not following? If it does error for what ever reason this will capture it and pass it through to the view controller which will then update the comment feed and display a message to the user

Reply:
Ah, but the error param is always `nil` here, as you ignore/discard it [here](https://github.com/rnystrom/GitHawk/pull/528/files#diff-5d794bd4eac3501e9bf9ef3658a7c4f4R190)

Reply:
Actually the value being discarded there is GitHub paging info, this endpoint can't be paged. There is technically a way to get network errors out of the response as we do on many requests, but we never show a detailed error regardless so knowing what error is not as important as knowing there is _an_ error.